tests/TESTS/common/serial_connection.py

```diff
@@ -45,6 +46,18 @@ def readline(self):
             log.error('Serial connection read error: {}'.format(se))
             return None
 
+    async def readline_async(self):
+        """
+        Asynchronously read line from serial port
+        :return: One line from serial stream
+        """
+        try:
+            output = await self.ser.readline_async()
+            return output
+        except asyncio.CancelledError:
+            log.error(f'readline_async() future cancelled')

```

Comment:
is this not fatal?

Reply:
It's a good question, but the answer is no. Look at the implementation of [readline_async](https://github.com/changyuheng/aioserial/blob/983ac4707a84aa18b8cc29697c04841503617252/packages/aioserial/aioserial.py#L121) in aioserial. It runs the pyserial readline in a background thread using [loop.run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor). Moreover, it wraps it in a try except block because the [Future](https://docs.python.org/3/library/asyncio-future.html) might be cancelled. However, if the future is cancelled, aioserial accepts it by [shield](https://docs.python.org/3/library/asyncio-task.html#shielding-from-cancellation)-awaiting _cancel_read_async. This cannot be cancelled because it is shielded. However, aioserial re-raises the error, so it makes sense to except it here too.

Reply:
so much more intuitive than threads